ReactiveCocoaTests/Swift/SignalProducerSpec.swift

```diff
@@ -154,83 +274,427 @@ class SignalProducerSpec: QuickSpec {
+
+						signal.observe(interrupted: {
+							interrupted = true
+						})
+
+						disposable.dispose()
+					}
+
+				expect(interrupted).to(beTruthy())
 			}
 
-			pending("should dispose of added disposables upon completion") {
+			it("should dispose of added disposables upon completion") {
+				let addedDisposable = SimpleDisposable()
+				var test: () -> () = { }
+
+				let producer = SignalProducer<Int, TestError>() { observer, disposable in
+					disposable.addDisposable(addedDisposable)
+					test = { _ in sendCompleted(observer) }
+				}
+
+				producer |> startWithSignal { _ in }
+				expect(addedDisposable.disposed).to(beFalsy())
+
+				test()
+				expect(addedDisposable.disposed).to(beTruthy())
 			}
 
-			pending("should dispose of added disposables upon error") {
+			it("should dispose of added disposables upon error") {
+				let addedDisposable = SimpleDisposable()
+				var test: () -> () = { }
+
+				let producer = SignalProducer<Int, TestError>() { observer, disposable in
+					disposable.addDisposable(addedDisposable)
+					test = { _ in sendError(observer, .Default) }
+				}
+
+				producer |> startWithSignal { _ in }
+				expect(addedDisposable.disposed).to(beFalsy())
+
+				test()
+				expect(addedDisposable.disposed).to(beTruthy())
 			}
 		}
 
 		describe("start") {
-			pending("should immediately begin sending events") {
+			it("should immediately begin sending events") {
+				let producer = SignalProducer<Int, NoError>(values: [1, 2])
+
+				var values: [Int] = []
+				var completed = false
+				producer.start(next: {
+					values.append($0)
+				}, completed: {
+					completed = true
+				})
+
+				expect(values).to(equal([1, 2]))
+				expect(completed).to(beTruthy())
 			}
 
-			pending("should send interrupted if disposed") {
+			it("should send interrupted if disposed") {
+				let producer = SignalProducer<(), NoError>.never
+
+				var interrupted = false
+				let disposable = producer.start(interrupted: {
+					interrupted = true
+				})
+
+				expect(interrupted).to(beFalsy())
+
+				disposable.dispose()
+				expect(interrupted).to(beTruthy())
 			}
 
 			pending("should release sink when disposed") {
+				weak var testSink: TestSink?
+
+				var dispose: () -> () = { }
+				var test: () -> () = {
+					let producer = SignalProducer<Int, NoError>.never
+					let sink = TestSink()
+					testSink = sink
+
+					let disposable = producer.start(sink)
+					dispose = { _ in disposable.dispose() }
+				}
+
+				test()
+				expect(testSink).toNot(beNil())
+
+				dispose()
+				expect(testSink).to(beNil())
 			}
 		}
 
 		describe("lift") {
 			describe("over unary operators") {
-				pending("should invoke transformation once per started signal") {
+				it("should invoke transformation once per started signal") {
+					let baseProducer = SignalProducer<Int, NoError>(values: [1, 2])
+
+					var counter = 0
+					let transform = { (signal: Signal<Int, NoError>) -> Signal<Int, NoError> in
+						counter += 1
```

Comment:
Double checked the stdlib, `+=` has no return value so doesn't confuse the compiler here.